Tangem/Common/Types/OrderedWeakObjectsCollection.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  OrderedWeakObjectsCollection.swift
+//  Tangem
+//
+//  Created by Sergey Balashov on 09.01.2023.
+//  Copyright © 2023 Tangem AG. All rights reserved.
+//
+
+import Foundation
+
+/// Use this class for saving a strong objects as weak
+public class OrderedWeakObjectsCollection<T> {
```

Comment:
может тут лучше сделать:
```suggestion
public class OrderedWeakObjectsCollection<T: AnyObject> {
```

и тогда можно всюду убрать касты в `AnyObject`

Reply:
✅

Reply:
чет я дурак, или лыжи не едут

![Screenshot 2023-01-10 at 10 35 09](https://user-images.githubusercontent.com/25799680/211489741-9d156b8d-3e26-45aa-b8cf-1542ba791941.png)

Reply:
чет я тупанул. Ты же передаешь в коллекцию протоколы, а не объекты, а протокол не может быть `AnyObject`, поэтому и ругается. тогда не получится обойтись без кастов

Reply:
сделал как оригинал был ))